src/mpi_native/mpi_native.cpp

```diff
@@ -47,28 +65,777 @@ int MPI_Init(int* argc, char*** argv)
+    (*win)->worldId = world.getId();
+    (*win)->size = size;
+    (*win)->dispUnit = disp_unit;
+    (*win)->rank = executingContext.getRank();
+    (*win)->basePtr = base;
+    world.createWindow(*win, (uint8_t*)base);
 
     return MPI_SUCCESS;
 }
 
-int MPI_Finalize()
+int MPI_Get_processor_name(char* name, int* resultlen)
+{
+    auto logger = faabric::util::getLogger();
+    logger->debug("MPI_Get_processor_name");
+
+    std::string host = faabric::util::getSystemConfig().endpointHost;
+    strncpy(name, host.c_str(), host.length());
+    *resultlen = host.length();
+
+    return MPI_SUCCESS;
+}
+
+int MPI_Win_get_attr(MPI_Win win,
+                     int win_keyval,
+                     void* attribute_val,
+                     int* flag)
 {
     auto logger = faabric::util::getLogger();
-    logger->debug("MPI_Finalize");
+    logger->debug("MPI_Win_get_attr");
+
+    *flag = 1;
+    if (win_keyval == MPI_WIN_BASE) {
+        *((void**)attribute_val) = win->basePtr;
+    } else {
+        if (win_keyval == MPI_WIN_SIZE) {
+            *((int*)attribute_val) = win->size;
+        } else if (win_keyval == MPI_WIN_DISP_UNIT) {
+            *((int*)attribute_val) = win->dispUnit;
+        } else {
+            throw std::runtime_error("Unrecofnised window attribute type " +
+                                     std::to_string(win_keyval));
+        }
+    }
+
+    return MPI_SUCCESS;
+}
+
+int MPI_Free_mem(void* base)
+{
+    auto logger = faabric::util::getLogger();
+    logger->debug("MPI_Free_mem");
+
+    return MPI_SUCCESS;
+}
+
+int MPI_Request_free(MPI_Request* request)
+{
+    notImplemented("MPI_Request_free");
+
+    return MPI_SUCCESS;
+}
+
+int MPI_Type_contiguous(int count, MPI_Datatype oldtype, MPI_Datatype* newtype)
+{
+    auto logger = faabric::util::getLogger();
+    logger->debug("MPI_Type_contiguous");
+
+    return MPI_SUCCESS;
+}
+
+int MPI_Type_commit(MPI_Datatype* type)
+{
+    auto logger = faabric::util::getLogger();
+    logger->debug("MPI_Type_commit");
+
+    return MPI_SUCCESS;
+}
+
+/*
+ * The API call returns a request ID that we can't capture with the function
+ * pointer trick. Thus we don't use it here.
+ */
+int MPI_Isend(const void* buf,
+              int count,
+              MPI_Datatype datatype,
+              int dest,
+              int tag,
+              MPI_Comm comm,
+              MPI_Request* request)
+{
+    auto logger = faabric::util::getLogger();
+    logger->debug("MPI_Isend {} -> {}", executingContext.getRank(), dest);
+
+    faabric::scheduler::MpiWorld& world = getExecutingWorld();
+    (*request) = (faabric_request_t*)malloc(sizeof(faabric_request_t));
+    int requestId = world.isend(
+      executingContext.getRank(), dest, (uint8_t*)buf, datatype, count);
+    (*request)->id = requestId;
+
+    return MPI_SUCCESS;
+}
+
+/*
+ * The API call returns a request ID that we can't capture with the function
+ * pointer trick. Thus we don't use it here.
+ */
```

Comment:
Not quite sure what you mean here but I'm not sure it's worth not doing something just to facilitate the function pointer idea.